tox.ini

```diff
@@ -23,6 +23,37 @@ commands = pytest --cov=caikit --cov-report=html {posargs:tests}
 ; Without this, sdist packaging is tested so that's a start.
 package=wheel
 
+[testenv:docs]
+recreate = True
+deps = 
+    -r {toxinidir}/docs-requirements.txt
+    alchemy-config>=1.1.1,<2.0.0
+    alchemy-logging>=1.0.4,<2.0.0
+    anytree>=2.7.0,<3.0
+    docstring-parser>=0.14.1,<0.16.0
+    grpcio-health-checking>=1.35.0,<2.0
+    grpcio>=1.35.0,<2.0
+    ijson>=3.1.4,<3.3.0
+    munch>=2.5.0,<4.0
+    protobuf>=3.19.0,<5
+    prometheus_client>=0.12.0,<1.0
+    py-grpc-prometheus>=0.7.0,<0.8
+    PyYAML>=6.0,<7.0
+    requests>=2.26.0,<3.0
+    semver>=2.13.0,<4.0
+    six>=1.16.0,<2.0.0
+    tqdm>=4.59.0,<5.0.0
+    py-to-proto>=0.2.0,<0.3.0
+    import-tracker>=3.1.5,<4
+changedir = docs
+
+; Disabled '-W' flag as warnings in the files
+; TOTO: Add back in once build warnings fixed
+commands =
+  sphinx-apidoc -f -o api/ ../caikit
+  sphinx-build -E -a -b html -T . _build/html
+skip_install = True
```

Comment:
I think all the deps are required in order for sphinx to correctly import the source and walk it to generate docs

Reply:
That's correct what @joerunde says. I didn't add them to `docs-requirements` because I thin they should be in a reqs file of its own for the project. However, they are defined in the [pyproject.toml](https://github.com/caikit/caikit/blob/main/pyproject.toml) and couldn't figure out how I could reference them.